### PR TITLE
 Add translations to process instance events v2.

### PR DESCRIPTION
**Pull request by star-in-the-jar**

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx [hunk 1]
```diff
@@ -9,31 +9,36 @@ import AccessTimeIcon from '@mui/icons-material/AccessTime';
...
 interface Props {
     processInstanceEvent: IProcessInstanceEvent;
 }
 
 const ProcessInstanceEventsDetailsHeader: VFC<Props> = ({ processInstanceEvent }) => {
+    const { translate } = useTranslations();
     const formattedStatus = convertToPascalCase(processInstanceEvent.status);
-    const [step, setStep] = useState<string>(processInstanceEvent.step)
+    const [translatedLabel, setTranslatedLabel] = useState(processInstanceEvent.step);
+    const translateKey = `Process.Details.Modeler.Actions.${((processInstanceEvent.step).split('.')).reduce((prev, curr) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: whole logic should be extracted to another const

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx [hunk 2]
```diff
@@ -9,31 +9,36 @@ import AccessTimeIcon from '@mui/icons-material/AccessTime';
...
+        if(curr){
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: is it possible to `curr` be falsy? imo `prev` can be undefined/empty

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx [hunk 3]
```diff
@@ -9,31 +9,36 @@ import AccessTimeIcon from '@mui/icons-material/AccessTime';
...
+            return prev.charAt(0).toUpperCase() + prev.slice(1) + '.' + curr.charAt(0).toUpperCase() + curr.slice(1);
+        } else {
+            return prev.charAt(0).toUpperCase() + prev.slice(1);
+        }
+    })}.Label`;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: there should be initial value as a 2nd argument in reduce

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 1 above)
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: add result type to reduce like `reduce<string>(`

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx [hunk 4]
```diff
@@ -9,31 +9,36 @@ import AccessTimeIcon from '@mui/icons-material/AccessTime';
...
 
     useEffect(() => {
-        if (checkIfKeyExists(processInstanceEvent.step)) {
-            setStep(translate(processInstanceEvent.step));
-        } else {
-            setStep(processInstanceEvent.step);
+        if (checkIfKeyExists(translateKey)) {
+            setTranslatedLabel(translate(translateKey));
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: I bet that we don't need useEffect here or is there a problem in lang switch? If not then you can create function that will be used inside of the Typography

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx [hunk 5]
```diff
@@ -19,10 +20,24 @@ const ActionLabelForm: VFC<Props> = ({ onSubmit }) => {
...
+    const translateKey = `Process.Details.Modeler.Actions.${(actionId.split('.')).reduce((prev, curr) => {
+        if(curr){
+            return prev.charAt(0).toUpperCase() + prev.slice(1) + '.' + curr.charAt(0).toUpperCase() + curr.slice(1);
+        } else {
+            return '.' + prev.charAt(0).toUpperCase() + prev.slice(1);
+        }
+    })}.Label`;
+    
+    const [translatedLabel, setTranslatedLabel] = useState(actionId);
+    console.log('translateKey: ', translateKey);
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx]: console log should be removed

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 1 above)
Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: imo you can rework `capitalizeFirstLetter` util function and use it everywhere, do not repeat yourself in each file

`export const capitalizeFirstLetter = (text: string, delimiter?: string, join?: string) => text
    .split(delimiter)
    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
    .join(join ?? '');`

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx [hunk 6]
```diff
@@ -19,10 +20,24 @@ const ActionLabelForm: VFC<Props> = ({ onSubmit }) => {
     const { element, action } = useBpmnFormContext();
     const [formState, setFormState] = useState({ editing: false, label: element.businessObject.label });
     const { translate } = useTranslations();
-    // using 'runbotics' xml field is a temporary solution
-    const translateKey = element.businessObject?.runbotics;
-    // @ts-ignore
-    const translatedLabel = translate(translateKey);
+    const actionId = element.businessObject?.actionId;
+    const translateKey = `Process.Details.Modeler.Actions.${(actionId.split('.')).reduce((prev, curr) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx]: logic repetition

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js [hunk 7]
```diff
@@ -64,11 +63,25 @@ export default class CustomRenderer extends BaseRenderer {
         });
 
         // svgClasses(text).add('djs-label');
-        const translatedLabel = translate(label.translateKey)
 
-        svgAppend(text, document.createTextNode(label.title ? label.title : translatedLabel));
+        const actionId = label.actionId;
+
+        if(actionId) {
+            const translateKey = `Process.Details.Modeler.Actions.${(actionId.split('.')).reduce((prev, curr) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js]: another repetition

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js]: each change in architecture and then you have to search for every usage like this

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: yep, curr is undefined if there's non-plural number of "words", e.g. 
.Browser.PrintToPdf.Variable -> three "words".

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: OUTDATED -> curr no longer used

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 3 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: outdated

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx (hunk 5 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx (hunk 6 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/ActionLabelForm.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js (hunk 7 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js]: done

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 4 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: outdated

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 4 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: outdated

File: runbotics/runbotics-orchestrator-ui/src/utils/text.ts [hunk 8]
```diff
@@ -1,7 +1,12 @@
-export const capitalizeFirstLetter = (text: string) => text.charAt(0).toUpperCase() + text.slice(1).toLowerCase();
+interface IText {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/utils/text.ts]: I in front of the interfaces is an antipattern, name it more specifically like `CapitalizeFirstLetterParams`

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx [hunk 9]
```diff
@@ -9,31 +9,23 @@ import AccessTimeIcon from '@mui/icons-material/AccessTime';
...
-        if (checkIfKeyExists(processInstanceEvent.step)) {
-            setStep(translate(processInstanceEvent.step));
-        } else {
-            setStep(processInstanceEvent.step);
-        }
-    }, [i18n.language])
-
+    const { translate } = useTranslations();
+    const formattedStatus = capitalizeFirstLetter({ text: processInstanceEvent.status, lowerCaseRest: true, delimiter: /_| / });
+    const translateKey = `Process.Details.Modeler.Actions.${capitalizeFirstLetter({ text: processInstanceEvent.step, lowerCaseRest: false, delimiter: '.', join: '.'})}.Label`;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: `lowerCaseRest` is false by default no need to include it, adjust all similar appearances

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js [hunk 10]
```diff
@@ -300,7 +307,7 @@ export default class CustomRenderer extends BaseRenderer {
             }
             return {
                 title: businessObject.label,
-                translateKey: translateKey,
+                actionId: actionId,
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js]: no need to double variable, JS will substitute value for the field like variable name

**Review by preludium — CHANGES_REQUESTED** [review]
watch you version number

File: runbotics/runbotics-orchestrator-ui/src/utils/text.ts (hunk 8 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/utils/text.ts]: roger that. Done

File: runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx (hunk 9 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/components/InfoPanel/ProcessInstanceEventsDetails/ProcessInstanceEventsDetailsHeader.tsx]: done

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js (hunk 10 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/extensions/custom/CustomRenderer.js]: done

**Review by preludium — APPROVED** [review]

**Pull request merged**